pkg/collector/annotations.go

```diff
@@ -43,22 +43,6 @@ func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
 	return annotations
 }
 
-// PodAnnotations return the spec annotations for OpenTelemetryCollector pod.
-func PodAnnotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
```

Comment:
I wonder why the pod annotations were separated from the others. Did you try to look it up in the history?

Reply:
I took a look at the history, but it seems this behavior was a try to fix a bug related to reconciling the collector once the configmap changes. However, I don't know why it wasn't simply added the sha256 onto the Deployment annotations and replicate them to the podAnnotations.

Reply:
https://github.com/open-telemetry/opentelemetry-operator/pull/670

LGTM to get rid of it

Reply:
Do you think we could have a situation in which pods should have a specific annotation not present on Deployment's annotations?